contracts/libraries/VaultLifecycle.sol

```diff
@@ -481,32 +481,42 @@ library VaultLifecycle {
         external
         view
         returns (
-            uint256 performanceFee,
-            uint256 managementFee,
+            uint256 performanceFeeInAsset,
+            uint256 managementFeeInAsset,
             uint256 vaultFee
         )
     {
         uint256 prevLockedAmount = vaultState.lastLockedAmount;
-        uint256 totalPending = vaultState.totalPending;
+
+        uint256 lockedBalanceSansPending =
+            currentLockedBalance.sub(vaultState.totalPending);
+
+        uint256 _performanceFeeInAsset;
+        uint256 _managementFeeInAsset;
+        uint256 _vaultFee;
 
         // Take performance fee and management fee ONLY if difference between
         // last week and this week's vault deposits, taking into account pending
         // deposits and withdrawals, is positive. If it is negative, last week's
         // option expired ITM past breakeven, and the vault took a loss so we
         // do not collect performance fee for last week
-        if (currentLockedBalance.sub(totalPending) > prevLockedAmount) {
-            performanceFee = currentLockedBalance
-                .sub(totalPending)
-                .sub(prevLockedAmount)
-                .mul(performanceFeePercent)
-                .div(100 * 10**6);
-            managementFee = currentLockedBalance
-                .sub(totalPending)
-                .mul(managementFeePercent)
-                .div(100 * 10**6);
-
-            vaultFee = performanceFee.add(managementFee);
+        if (lockedBalanceSansPending > prevLockedAmount) {
+            _performanceFeeInAsset = performanceFeePercent > 0
+                ? lockedBalanceSansPending
+                    .sub(prevLockedAmount)
+                    .mul(performanceFeePercent)
+                    .div(100 * Vault.FEE_MULTIPLIER)
+                : 0;
+            _managementFeeInAsset = managementFeePercent > 0
+                ? currentLockedBalance.mul(managementFeePercent).div(
+                    100 * Vault.FEE_MULTIPLIER
+                )
```

Comment:
I see we're missing `.sub(totalPending)` for both fees, is there a reason why?

Reply:
because lockedBalanceSansPending already includes that

Reply:
Okay I see it, but we should be using `lockedBalanceSansPending` instead of `currentLockedBalance` on line 511